File: app/src/main/java/com/jerson/hcdc_portal/database/GradeDao.java
package com.jerson.hcdc_portal.database;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;

import com.jerson.hcdc_portal.model.GradeModel;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Flowable;

@Dao
public interface GradeDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertGrade(List<GradeModel> grade);

    @Query("select * from grade where link_id=:link_id")
    Flowable<List<GradeModel>> getGrade(int link_id);

    @Query("delete from grade where link_id=:link_id")
    Completable deleteGradeData(int link_id);

    // link
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Completable insertGradeLink(List<GradeModel.Link> gradeLink);

    @Query("select * from gradelink")
    Flowable<List<GradeModel.Link>> getGradeLink();

    @Query("delete from gradelink")
    Completable deleteGradeLink();
}

File: app/src/main/java/com/jerson/hcdc_portal/data/local/dao/GradeDao.kt
package com.jerson.hcdc_portal.data.local.dao

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.jerson.hcdc_portal.domain.model.Grade
import kotlinx.coroutines.flow.Flow
@Dao
interface GradeDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    suspend fun upsertGrade(grades: List<Grade>)

    @Query("delete from grades where termId = :termId")
    suspend fun deleteAllGrades(termId:Int)
    @Query("delete from grades")
    suspend fun deleteAllGrades()

    @Query("select * from grades where termId = :termId")
    fun getGrades(termId:Int): Flow<List<Grade>>
}